lib/onelogin/ruby-saml/response.rb

```diff
@@ -51,6 +53,21 @@ def initialize(response, options = {})
 
         @response = decode_raw_saml(response)
         @document = XMLSecurity::SignedDocument.new(@response, @errors)
+
+        return unless assertion_encrypted?
```

Comment:
@luisvm, I think that your proposal turn the code more complex..and this is maybe what we are trying to avoid..